### PR TITLE
Add hamming distance calculation to bloom filters

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

File: Godeps/_workspace/src/github.com/steakknife/hamming/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Copyright (c) 2014 Barry Allard
+
+MIT license
```
Comment (jbenet) [inline Godeps/_workspace/src/github.com/steakknife/hamming/README.md]: the code doesn't bear an actual `LICENSE` file. https://github.com/steakknife/hamming/pull/3

**Comment by jbenet** [comment]
LGTM other than `LICENSE` concern above. @whyrusleeping ?

File: Godeps/_workspace/src/github.com/steakknife/hamming/README.md (hunk 1 above)
Comment (steakknife) [inline Godeps/_workspace/src/github.com/steakknife/hamming/README.md]: @jbenet Tyrannical, bureaucratic "standards" people exist to foist unnecessary bikeshedding FUD and work on others while contributing no real value.  I refuse to set this precedent unless you have a really good reason.

Comment (jbenet) [inline Godeps/_workspace/src/github.com/steakknife/hamming/README.md]: This is not bikeshedding. I've checked this with lawyers multiple times. As things stand today, it is still required to have the full text of licenses along with the relevant code base. 




I already did the work for you. If you can't press one button, fine. We'll add it ourselves here. You're foisting much more work unto us -- every time we vendor -- but I don't care about arguing you on this



—

File: blocks/bloom/filter.go [hunk 2]
```diff
@@ -100,3 +102,23 @@ func (f *filter) Merge(o Filter) (Filter, error) {
...
+		return 0, errors.New("Unsupported filter type")
+	}
+
+	if len(f.filter) != len(casfil.filter) {
+		return 0, errors.New("filter lengths must match!")
+	}
+
+	acc := 0
+
+	// xor together
```
Comment (whyrusleeping) [inline blocks/bloom/filter.go]: whats being xor'ed?

**Comment by whyrusleeping** [comment]
this LGTM, just curious about the usecase

**Comment by jbenet** [comment]
I pushed a commit with the newly vendored license. will merge when tests pass.

**Pull request merged**